src/main/java/seedu/smarthomebot/logic/commands/OnCommand.java

```diff
@@ -14,55 +17,74 @@
 import static seedu.smarthomebot.commons.Messages.MESSAGE_INVALID_TEMPERATURE_AC;
 import static seedu.smarthomebot.commons.Messages.MESSAGE_INVALID_FAN_SPEED;
 
+//@@author Ang_Cheng_Jun
 /**
- * Represent the command to turn on the appliance.
+ * Represent the command to turn on the appliance(s).
  */
 public class OnCommand extends Command {
 
     public static final String COMMAND_WORD = "on";
     public static final String MESSAGE_USAGE = "Switch ON Appliance: \n\t\t a. " + COMMAND_WORD
             + " [APPLIANCE_NAME] \n\t\t b. " + COMMAND_WORD + " [APPLIANCE_NAME] p/[PARAMETER] \n\t\t c. "
             + COMMAND_WORD + " [LOCATION_NAME]";
+
     private static final String APPLIANCE_TYPE = "appliance";
     private static final String LOCATION_TYPE = "location";
-    private final String key;
+    private final String argument;
     private final String parameter;
 
-    public OnCommand(String key, String parameter) {
-        this.key = key;
+    public OnCommand(String argument, String parameter) {
+        this.argument = argument;
         this.parameter = parameter;
     }
 
     @Override
     public CommandResult execute() {
-        String onByType = APPLIANCE_TYPE;
-        // To check if any of the appliances contains the name of the location
-        ArrayList<Appliance> filterApplianceList =
-                (ArrayList<Appliance>) applianceList.getAllAppliance().stream()
-                        .filter((s) -> s.getLocation().equals(this.key))
-                        .collect(toList());
-
-        // if list is empty
-        if (!filterApplianceList.isEmpty()) {
-            onByType = LOCATION_TYPE;
-        }
-        switch (onByType) {
-        case (APPLIANCE_TYPE):
-            return onByAppliance();
-        case (LOCATION_TYPE):
-            return onByLocation();
-        default:
-            return new CommandResult("");
+        try {
+            String onByType = APPLIANCE_TYPE;
+            ArrayList<Appliance> filterApplianceList =
```

Comment:
Should we explain what this stream does? I feel it could be hard for future developers to understand this block. Maybe in the form of comment